operator/webhooks/pod_mutating_webhook.go

```diff
@@ -141,12 +143,23 @@ func (a *PodMutatingWebhook) isAppAnnotationPresent(pod *corev1.Pod) (bool, erro
 		return true, nil
 	}
 
-	pod.Annotations[common.AppAnnotation] = pod.Annotations[common.WorkloadAnnotation]
+	if len(pod.Annotations) == 0 {
+		pod.Annotations = make(map[string]string)
+	}
+	pod.Annotations[common.AppAnnotation], _ = getLabelOrAnnotation(pod, common.WorkloadAnnotation, common.K8sRecommendedWorkloadAnnotations)
 	return false, nil
 }
 
 func (a *PodMutatingWebhook) calculateVersion(pod *corev1.Pod) string {
 	name := ""
+
+	if len(pod.Spec.Containers) == 1 {
+		image := strings.Split(pod.Spec.Containers[0].Image, ":")
+		if len(image) > 0 && image[1] != "" && image[1] != "latest" {
```

Comment:
if the separator doesn't exist, `strings.split` always return a slice of a single element
```suggestion
		if len(image) > 1 && image[1] != "" && image[1] != "latest" {
```